flink-runtime/src/main/java/org/apache/flink/runtime/heartbeat/HeartbeatManagerImpl.java

```diff
@@ -106,8 +107,8 @@ Executor getExecutor() {
 		return heartbeatListener;
 	}
 
-	Collection<HeartbeatManagerImpl.HeartbeatMonitor<O>> getHeartbeatTargets() {
-		return heartbeatTargets.values();
+	Collection<Map.Entry<ResourceID, HeartbeatMonitor<O>>> getHeartbeatTargets() {
```

Comment:
What about adding a `getTargetID()` to the `HeartbeatMonitor`? Then we would not have to expose, even though it is an internal method, that we use a `Map` to store the heartbeat targets. We could then keep the return type `Collection<HeartbeatMonitor>`.

Reply:
yes that is a better solution.